src/tests/Common/XUnitWrapperLibrary/TestFilter.cs

```diff
@@ -191,30 +210,83 @@ public bool ShouldRunTest(string fullyQualifiedName, string displayName, string[
         if (shouldRun)
         {
             // Test stripe, if true, then report success
-            return ((System.Threading.Interlocked.Increment(ref _shouldRunQuery)) % _stripeCount) == _stripe;
+            return ((Interlocked.Increment(ref _shouldRunQuery)) % _stripeCount) == _stripe;
         }
         return false;
     }
 
-    public static HashSet<string> LoadTestExclusionList()
+    public string GetTestExclusionReason(string testDisplayName)
+    {
+        if (_testExclusionTable is null)
+            return string.Empty;
+
+        string trueDisplayName = testDisplayName.Replace("\\", "/");
+
+        return _testExclusionTable.ContainsKey(trueDisplayName)
+               ? _testExclusionTable[trueDisplayName]
+               : string.Empty;
+    }
+
+    // GH dotnet/runtime issue #91562: Some tests are purposefully not run for a number
+    // of reasons. They are specified in src/tests/issues.targets, along with a brief
+    // explanation on why they are skipped inside an '<Issue>' tag.
+    //
+    // When building any test or test subtree, if any of the tests built matches an entry
+    // in issues.targets, then the exclusions list file ($CORE_ROOT/TestExclusions.txt is
+    // the default) is updated by adding a new a comma-separated entry:
```

Comment:
IIRC they are actually semicolon / CRLF-separated.

Reply:
Hmm, maybe I just misunderstood this and the comma separation is a new addition of yours, I continue looking...

Reply:
Oh no, this is the `TestExclusions.txt` file. It used to be just one test per line, and I modified it to also include the reason with a comma. As for line endings, we're now using the platform's one (i.e. Linux/MacOS LF and Windows CRLF).